### PR TITLE
Allow Magento HTTP Backend when configured

### LINKED ISSUE #2422: [feature]: Allow Magento HTTP Backend (without SSL) when configured
**Is your feature request related to a problem? Please describe.**
In case when you want to set - "**MAGENTO_BACKEND_URL**" to URL without SSL (e.g. http, not a https) - you would get an error - "http is not supported".

I think such possibility would be great for cases when you are running "**pwa-studio**" as separate docker container on the same host or within same kubernetes cluster together with other magento-related containers. 

In such case requests are made within private network or even on the same host. So, requiring https would mean that we still need to use SSL reverse/offload-proxies internally to communicate between the containers.

Example:
- Kubernetes cluster.
- Ingress used to offload client requests separately to PWA/Magento
- Internally Magento's Varnish available as http endpoint, so PWA could communicate directly to Varnish, avoiding additional step to proxy data via Ingress.

**Describe the solution you'd like**
I have debugged this and found quite simple solution. Will create PR soon.
Commit is here: https://github.com/IgorVitol/pwa-studio/commit/554f24e368e06032c0137e607132ff529dfa893f

**Describe alternatives you've considered**
No alternatives

**Additional context**
No additional context

**Please let us know what packages this feature is in regards to:**
- [ ] `venia-concept`
- [ ] `venia-ui`
- [x] `pwa-buildpack`
- [ ] `peregrine`
- [ ] `pwa-devdocs`
- [x] `upward-js`
- [ ] `upward-spec`
- [ ] `create-pwa`

### PR DESCRIPTION
**Pull request by IgorVitol**
## Description

Small changes in order to allow http backend url - "**MAGENTO_BACKEND_URL**"

## Related Issue

Closes #2422

### Specification

Changes to next packages:
- `upward-js`
- `pwa-buildpack`

### Verification Steps

1. Set http backend URL for **MAGENTO_BACKEND_URL**
2. Build project & run server
3. Verify that it works - catalog, images, place order.
4. Do the same for https.

## Checklist

* I have updated the documentation accordingly, if necessary.
[ ] I have added tests to cover my changes, if necessary.

### CODE REVIEW

**Comment by IgorVitol** [comment]
@Jordaneisenburger Could you pls take a look at this issue - what next steps required after PR creation?

**Comment by devops-pwa-codebuild** [comment]
**Performance Test Results**

The following fails have been reported by WebpageTest. These numbers indicates a possible performance issue with the PR which requires further manual testing to validate.

https://pr-2423.pwa-venia.com : LH Performance Expected 0.85 Actual 0.55, LH Best Practices Expected 1 Actual 0.92
https://pr-2423.pwa-venia.com/venia-tops.html : LH Performance Expected 0.75 Actual 0.34, LH Best Practices Expected 1 Actual 0.92
https://pr-2423.pwa-venia.com/valeria-two-layer-tank.html : LH Performance Expected 0.8 Actual 0.51, LH Accessibility Expected 0.9 Actual 0.89, LH Best Practices Expected 1 Actual 0.92

**Comment by tjwiebell** [comment]
@IgorVitol - Thank you for the contribution. We've groomed the linked issue and have moved this up our backlog queue to be reviewed. Within the next week or so we should be in touch if we have any feedback.

File: packages/pwa-buildpack/lib/Utilities/graphQL.js [hunk 1]
```diff
@@ -1,22 +1,26 @@
...
+    const targetURL = new URL('graphql', process.env.MAGENTO_BACKEND_URL);
+    const httpsOptions = {};
+    if (targetURL.protocol === 'https:') {
+        httpsOptions.agent = httpsAgent;
+    }
     return fetch(
-        new URL('graphql', process.env.MAGENTO_BACKEND_URL).toString(),
-        {
+        targetURL.toString(),
+        Object.assign(httpsOptions, {
```
[the remark this answers is not in the document]
Comment (tjwiebell) [reply packages/pwa-buildpack/lib/Utilities/graphQL.js]: Great idea! Changed in 0046e26.

**Comment by dpatil-magento** [comment]
@zetlen build/deploy is working with `http` backend now but images still does not load. Should we expect them to load?

cc - @tjwiebell 

image

**Comment by tjwiebell** [comment]
> @zetlen build/deploy is working with `http` backend now but images still does not load. Should we expect them to load?
> 
> cc - @tjwiebell

Images fail to load when using a secured storefront with an insecure backend, so browser default CSP is kicking in, which appears to be very secure by default. We just want to confirm if this setup is something we do need to support; do we need to allow mixed content when using insecure backend?

**Comment by dpatil-magento** [comment]
QA Approved.

**Comment by IgorVitol** [comment]
@tjwiebell Thanks for the patch!

> @zetlen build/deploy is working with `http` backend now but images still does not load. Should we expect them to load?
> 
> cc - @tjwiebell
> 
> image

Here is two things to keep in mind:
1. If you want to use it with pure http backend, the only single way to do it - keep using "**IMAGE_OPTIMIZING_ORIGIN: onboard**" optimization. Otherwise, while using image origin: **backend** - it will show you http/https protocol missmatch.

2. This http-backend support is useful when you have your backend & PWA Studio at https urls anyway, but using SSL offloading on request flow. So, internally they could communicate via internal http url, but globally requests will be served via https load balancers.
You could refer to example shown in #2422

**Comment by dpatil-magento** [comment]
Thanks @IgorVitol , images does load with `onboard` config.

**Pull request merged**